### DISCUSSION TITLE
How to use cudnn and tensorrt

### DISCUSSION THREAD

**Discussion by hofbi** [bazel-contrib/rules_cuda#244] [Q&A] [answered]
Is there a recommended way on how to use `cudnn` and `tensorrt`? Is this something that should be supported by rules_cuda?

**Comment by cloudhan** [comment]
They are not in cuda toolkit, aka, seperate nvidia product build on top of cuda toolkit. So you need to wrap them by yourself.

**Comment by hofbi** [comment]
Would it make sense to put some documentation for a best practice way here into the repo/examples? I expect people searching for this might end up here.

**Comment by cloudhan** [comment, marked as answer]
[third_party.zip](https://github.com/bazel-contrib/rules_cuda/files/15437150/third_party.zip)

```starlark
# In your WORKSPACE.bazel
load("//third_party/cudnn:deps.bzl", "cudnn_dependencies")
cudnn_dependencies()
```

The code is extracted from a personal private repo and is rather outdated, you might want to tweak it to fit your need. Just use it as public domain, WTFPL if you want.

For trt, you should follow the similar way to import the library.

**Comment by wep21** [comment]
I created [rules_cudnn](https://github.com/wep21/rules_cudnn) based on @cloudhan 's code. I'm not familiar with bazel, so I would appreciate it if someone could give me feedback.

**Comment by ZhenshengLee** [comment]
there is [google-ml-infra/rules_ml_toolchain](https://github.com/google-ml-infra/rules_ml_toolchain) that create Hermetic CUDA, CUDNN, NCCL and NVSHMEM

doc is in https://github.com/google-ml-infra/rules_ml_toolchain/blob/main/third_party/gpus/hermetic_toolkits.md
